src/extension.ts

```diff
@@ -345,7 +349,95 @@ function showUpdateMessage(version: string) {
     });
 }
 
-function copyToClipboard(code: string) {
-    copyPaste.copy(code);
-    window.showInformationMessage("Step Implementation copied to clipboard");
+function putStubImplementationInFileContent(cwd, filePath, stepTexts) {
+    return new Promise((resolve,reject) => {
+        clients.get(cwd).sendRequest("gauge/putStubImpl", {implementationFilePath: filePath, stepTexts: stepTexts}, new CancellationTokenSource().token).then(
+            (val: {fileName: string, fileContent: string}) => {
+                resolve(val.fileContent)
+            },
+            (reason) => {
+                reject(reason)
+            }
+        );
+    })
+}
+
+function writeToFileInTextEditor(filePath, contents) {
+    if (!fs.existsSync(filePath)) {
+        fs.writeFileSync(filePath, '')
+    }
+    workspace.openTextDocument(filePath).then(
+        (doc) => {
+            window.showTextDocument(doc).then(
+                (document) => {
+                    var range = new Range(new Position(0, 0), new Position(doc.lineCount,0))
+                    document.edit(editBuilder => editBuilder.replace(range, contents))
+                }
+            )
+        }
+    )
+}
+
+function appendToFile(fileName) {
+    let cwd = workspace.getWorkspaceFolder(window.activeTextEditor.document.uri).uri.fsPath;
+    var position = window.activeTextEditor.selection.start;
+    var text = window.activeTextEditor.document.lineAt(position.line).text.slice(1);
+    var texts = [].concat(text);
+    if (fileName == NEW_FILE_DISPLAY_MESSAGE) {
+        window.showInputBox().then((val) => {
+            putStubImplementationInFileContent(cwd, val, texts).then(
+                (content) => {
+                    writeToFileInTextEditor(val, content)
+                },
+                (reason) => {
+                    window.showErrorMessage("Could Not Write to file because "+ reason);
+                }
+            )
+        })
+    } else if (fileName == COPY_TO_CLIPBOARD_DISPLAY_MESSAGE) {
+        putStubImplementationInFileContent(cwd, "", texts).then(
+            (content) => {
+                copyPaste.copy(content)
+                window.showInformationMessage("Step Implementation copied to clipboard")
+            },
+            (reason) => {
+                window.showErrorMessage("Step Implementation could not be copied to clipboard because "+ reason);
+            }
+        )
+    } else {
+        putStubImplementationInFileContent(cwd, fileName, texts).then(
+            (content) => {
+                writeToFileInTextEditor(fileName, content)
```

Comment:
A list of workspaceEdits should be applied. Explicitly writing to the files is not a good approach.